packages/preload-webview/src/webview-preload.ts

```diff
@@ -0,0 +1,147 @@
+  #webviewId: string;
+  #webviewInfo: WebviewInfo | undefined;
+  #domLoaded: boolean = false;
+  #acquiredApi: boolean = false;
+
+  constructor(webviewId: string) {
+    this.#webviewId = webviewId;
+  }
+
+  protected decodeError(error: ErrorMessage) {
+    const e = new Error(error.message);
+    e.name = error.name;
+    Object.assign(e, error.extra);
+    return e;
+  }
+
+  protected async ipcInvoke(channel: string, ...args: unknown[]): Promise<unknown> {
+    const { error, result } = await ipcRenderer.invoke(channel, ...args);
+    if (error) {
+      throw this.decodeError(error);
+    }
+    return result;
+  }
+
+  protected postWebviewMessage(message: unknown) {
+    this.ipcInvoke('webviewRegistry:post-message', this.#webviewInfo?.id, message).catch((error: unknown) =>
+      console.error('Error while posting message', error),
+    );
+  }
+
+  protected changeContent() {
+    if (!this.#webviewInfo) {
+      return;
+    }
+    if (!this.#domLoaded) {
+      return;
+    }
+    let webviewHtmlContent = '';
+    if (this.#webviewInfo) {
+      webviewHtmlContent = this.#webviewInfo.html;
+    }
+    // use a timeout to perform the update
+    setTimeout(() => {
+      const webviewContentHtml = new DOMParser().parseFromString(webviewHtmlContent, 'text/html');
+
+      webviewContentHtml.documentElement.style.setProperty('font-family', 'Montserrat');
+
+      const htmlContent = '<!DOCTYPE html>\n' + webviewContentHtml.documentElement.outerHTML;
+
+      document.open();
+      document.write(htmlContent);
+      document.close();
+    }, 0);
+  }
+
+  // build the function that will be exposed to the webview for getState/postMessage/setState
+  protected buildApi(): unknown {
+    return () => {
+      // initialize the state from the webview
+      let state: unknown = this.#webviewInfo?.state ?? {};
+      if (this.#acquiredApi) {
+        throw new Error('An instance of the Podman Desktop API has already been acquired');
+      }
+      // can only be called once;
+      this.#acquiredApi = true;
+      return Object.freeze({
+        getState: () => {
+          return state;
+        },
+        postMessage: (msg: unknown) => {
+          return this.postWebviewMessage({ command: 'onmessage', data: msg });
+        },
+        setState: async (newState: unknown) => {
+          state = newState;
+          // need to send back the state to the main process
+          this.ipcInvoke('webviewRegistry:update-state', this.#webviewInfo?.id, newState).catch((error: unknown) => {
+            console.error('Error while updating the state', error);
+          });
+        },
+      });
+    };
+  }
+
+  protected getWebviews(): Promise<WebviewInfo[]> {
+    return this.ipcInvoke('webviewRegistry:listWebviews') as Promise<WebviewInfo[]>;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  protected ipcRendererOn(channel: string, listener: (event: IpcRendererEvent, ...args: any[]) => void) {
```

Comment:
Can't we use `unknown` here ?

Reply:
if it works for you I'll come back with a follow-up PR I think I had an issue initially with the spread arguments

Reply:
fixing there: https://github.com/containers/podman-desktop/pull/5603